src/app/components/tags.ts

```diff
@@ -0,0 +1,36 @@
+import { BaseComponent, type ElementFnProps } from '@components/base-component';
+
+export const span = (props: ElementFnProps<HTMLElement>, ...children: BaseComponent[]) =>
+  new BaseComponent({ ...props, tag: 'span' }, ...children);
+
+export const main = (props: ElementFnProps, ...children: BaseComponent[]) =>
+  new BaseComponent({ ...props, tag: 'main' }, ...children);
```

Comment:
Functions main, label, and iconFromCode create BaseComponent without specifying a generic type. Consider adding explicit types for clarity.
```suggestion
export const main = (props: ElementFnProps<HTMLElement>, ...children: BaseComponent[]) =>
  new BaseComponent<HTMLElement>({ ...props, tag: 'main' }, ...children);

```